File: backend/src/main/java/com/enterprise/meeting/entity/Meeting.java
package com.enterprise.meeting.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "meetings")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Meeting {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 256)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "meeting_time")
    private LocalDateTime meetingTime;

    @Column(length = 256)
    private String location;

    @Column(name = "duration_minutes")
    private Integer durationMinutes;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_reviewer")
    private User assignedReviewer;

    @Column(nullable = false, length = 32)
    private String state;

    @Column(name = "ai_job_id", length = 128)
    private String aiJobId;

    @Column(name = "ai_retry_count")
    private Integer aiRetryCount = 0;

    @Column(name = "ai_error", columnDefinition = "TEXT")
    private String aiError;

    @Column(name = "state_metadata", columnDefinition = "TEXT")
    private String stateMetadata;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: android/app/src/main/java/com/enterprise/meeting/domain/model/Meeting.kt
package com.enterprise.meeting.domain.model

data class Meeting(
    val id: Long,
    val title: String,
    val description: String?,
    val meetingTime: Long,
    val location: String?,
    val durationMinutes: Int?,
    val createdBy: Long,
    val createdByName: String?,
    val assignedReviewer: Long?,
    val assignedReviewerName: String?,
    val state: MeetingState,
    val stateMetadata: Map<String, String> = emptyMap(),
    val aiRetryCount: Int = 0,
    val aiError: String?,
    val createdAt: Long,
    val updatedAt: Long
)

enum class MeetingState {
    PENDING,
    AI_PROCESSING,
    PENDING_REVIEW,
    REJECTED,
    PENDING_PUBLISH,
    PUBLISHED,
    ARCHIVED;

    val displayName: String
        get() = when (this) {
            PENDING -> "执行中"
            AI_PROCESSING -> "AI处理中"
            PENDING_REVIEW -> "待审核"
            REJECTED -> "已驳回"
            PENDING_PUBLISH -> "待发布"
            PUBLISHED -> "已发布"
            ARCHIVED -> "已归档"
        }
}
